### PR TITLE
sync versions script

### PR DESCRIPTION
**Pull request by perf2711**
This script helps to synchronize all versions of workspace packages in all workspaces.

For example, if `sdk-core` version changes, instead of going to every package and updating the version manually, one has to only run `npm run syncVersions` and it will take care of everything.

Example:
1. update sdk-core from 0.0.5 to 0.0.6
2. run `npm run syncVersions`
Output:
```
> @backtrace-labs/javascript@0.0.1 syncVersions
> ts-node ./scripts/syncVersions.ts

[@backtrace-labs/sdk-core] - no changes
[@backtrace-labs/browser] - updated @backtrace-labs/sdk-core from ^0.0.5 to ^0.0.6 in dependencies
[@backtrace-labs/react] - updated @backtrace-labs/sdk-core from ^0.0.5 to ^0.0.6 in dependencies
[@backtrace-labs/node] - updated @backtrace-labs/sdk-core from ^0.0.5 to ^0.0.6 in dependencies
[@backtrace-labs/nestjs] - no changes
[@backtrace-labs/react-native] - updated @backtrace-labs/sdk-core from ^0.0.5 to ^0.0.6 in dependencies
[@backtrace-labs/sourcemap-tools] - no changes
[@backtrace-labs/javascript-cli] - no changes
[@backtrace-labs/rollup-plugin] - no changes
[@backtrace-labs/webpack-plugin] - no changes
[@backtrace-labs/vite-plugin] - no changes
```

Also, if you want to sync versions of specific packages, you can pass paths to package.jsons:
```
> npm run syncVersions -- packages/node/package.json packages/browser/package.json 

> @backtrace-labs/javascript@0.0.1 syncVersions
> ts-node ./scripts/syncVersions.ts packages/node/package.json packages/browser/package.json

[@backtrace-labs/node] - updated @backtrace-labs/sdk-core from ^0.0.5 to ^0.0.6 in dependencies
[@backtrace-labs/browser] - updated @backtrace-labs/sdk-core from ^0.0.5 to ^0.0.6 in dependencies
```

### CODE REVIEW

File: scripts/syncVersions.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    const args = process.argv.slice(2);
+    const packageJsonPathsToSync = args.length ? args : workspacePackageJsonPaths;
+
+    const currentVersions = workspacePackageJsons.reduce((obj, pj) => {
+        obj[pj.name] = pj.version;
+        return obj;
+    }, {} as Record<string, string>);
+
+    for (const packageJsonPath of packageJsonPathsToSync) {
```
Comment (konraddysput) [inline scripts/syncVersions.ts:102]: can we run `npm install` in the end if there is any change? 

we're using `npm ci` which requires from us to keep our package-lock.json up to date. We can also automate this by using this tool. 

WDYT?

File: scripts/syncVersions.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return false;
+    }
+
+    const currentVersion = deps[name];
+    if (!currentVersion) {
+        return false;
+    }
+
+    if (currentVersion !== newVersion) {
+        deps[name] = newVersion;
```
Comment (konraddysput) [inline scripts/syncVersions.ts:62]: shouldn't this be `^${newVersion}` instead? Otherwise our concept with sdk-core won't work

**Review by konraddysput — COMMENTED** [review]
Left some comments before we land the tool. WDYT?

File: scripts/syncVersions.ts (hunk 2 above)
Comment (perf2711) [reply scripts/syncVersions.ts:62]: it is, see line 75

File: scripts/syncVersions.ts (hunk 1 above)
Comment (perf2711) [reply scripts/syncVersions.ts:102]: I wouldn't do that. Even if we decide that, it should be a part of scripts in package.json: 
```
ts-node ./scripts/syncVersions.ts && npm install
```

File: scripts/syncVersions.ts (hunk 2 above)
Comment (konraddysput) [reply scripts/syncVersions.ts:62]: ah ok. thx

**Review by konraddysput — APPROVED** [review]

**Review by adamcronin42 — APPROVED** [review]

**Pull request merged**